src/ess/nmx/mtz_io.py

```diff
@@ -0,0 +1,202 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+from typing import NewType, Optional
+
+import gemmi
+import numpy as np
+import pandas as pd
+import sciline as sl
+
+# Index types for param table.
+MTZFileIndex = NewType("MTZFileIndex", int)
+"""The index of the mtz file when iterating over multiple mtz files."""
+
+# User defined or configurable types
+MTZFilepath = NewType("MTZFilepath", str)
+"""Path to the mtz file"""
+LambdaBinSize = NewType("LambdaBinSize", int)
```

Comment:
```suggestion
LambdaBinCount = NewType("LambdaBinCount", int)
```
?
`Size` makes me think it is the bind width.

Reply:
Actually it's not used in this scope so I removed it for now. I'll apply it in along with the relevant changes. Thank you!